heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -396,11 +407,15 @@ private void accept(final Consumer<MetricBackend> op) {
         public void resolved(final Pair<Series, FetchData> result) throws Exception {
             final FetchData f = result.getRight();
             requestErrors.addAll(f.getErrors());
-
-            for (final MetricCollection g : f.getGroups()) {
-                g.updateAggregation(session, result.getLeft().getTags(),
-                    ImmutableSet.of(result.getLeft()));
-                dataInMemoryReporter.reportDataNoLongerNeeded(g.size());
+            try {
+                for (final MetricCollection g : f.getGroups()) {
+                    g.updateAggregation(session, result.getLeft().getTags(),
+                        ImmutableSet.of(result.getLeft()));
+                    dataInMemoryReporter.reportDataNoLongerNeeded(g.size());
+                }
+            } catch (QuotaViolationException e) {
+                // Seems resolved() should not throw exception
+                log.debug("Aggregation quota violated", e);
```

Comment:
Hm, at most as many errors as there are fetches under the current design? But for now I'm ok with this due to them not being propagated below regardless.